### ISSUE TITLE
Safari 5 does not work with autoplay

### ISSUE THREAD

**Issue by doublex** [videojs/video.js#2825]
Videojs 5.2.1, Safari 5.1.7
If   ```autoplay``` is enabled, the video does not work with Safari 5 (some users reported also problems with safari 6).

The problem is line 9650 in "video.js".
Error message: "TypeError: Unable to delete property."

If the line ```delete this.tag.poster;``` is enclosed in a try/catch block the player works without problems.

````
    // Chrome and Safari both have issues with autoplay.
    // In Safari (5.1.1), when we move the video element into the container div, autoplay doesn't work.
    // In Chrome (15), if you have autoplay + a poster + no controls, the video gets hidden (but audio plays)
    // This fixes both issues. Need to wait for API, so it updates displays correctly
    if (this.tag && this.options_.autoplay && this.paused()) {
      delete this.tag.poster; // Chrome Fix. Fixed in Chrome v16.
      this.play();
    }
````

Demo:
http://www.doppelbauer.name/video-js-5.2.1/examples/simple-embed/

**Comment by richardbushell** [comment]
I am sure that this isn't limited to Safari 5. I think it's also present in any Safari version up to and including Safari 9 (latest).
See the post:
https://github.com/videojs/video.js/issues/2815
Seems there's a problem between Safari and VJS if there's a POSTER included.
Also posted here:
https://github.com/videojs/video.js/issues/2807
https://github.com/videojs/videojs-contrib-dash/issues/22
Again, I don't think the problem is specific to Autoplay, but rather to any Safari vdeo with the Poster set.

**Comment by doublex** [comment]
Playing a video also fails on the device ```Mozilla/5.0 (iPad; CPU OS 6_1_3 like Mac OS X) AppleWebKit/536.26 (KHTML, like Gecko) Version/6.0 Mobile/10B329 Safari/8536.25``` in this line: ```delete this.tag.poster;```

**Comment by sprice** [comment]
@richardbushell Have you reproduced the poster problem with stock video.js and not dash? As far as I can tell, the poster issue is a dash.js problem, and not a video.js problem.

**Comment by Soviut** [comment]
This might be due to their power save mode where it will not play a video if it's not "front and center". (http://www.tekrevue.com/tip/disable-safari-power-saver-os-x/) The rules I've found that trigger this are:

- video is obstructed by other html elements (an overlay or a div used as a poster, for example)
- video X or Y coordinates are less than 0px (they're offscreen left or top)
- video X or Y coordinates are greater than 3000px (this might change depending on screen size, my numbers were based on a 13" macbook screen in the 1440 x 900 range)

**Comment by richardbushell** [comment]
@soviut - now that's a very interesting theory! The problems we (and others) have been reporting with Mac El Capitan & Safari 9 do seem to revolve around the POSTER attribute being set. We don't get initial playback, just the spinner, unless we nudge the time cursor along, and we've also had issues with the Poster re-displaying and remaining on screen after seeking sometimes (presumably in front of the video element as we can hear audio). Unfortunately, we currently have no Macs to test it on, we occasionally have to borrow one to conduct our tests, but would be great if anyone else wants to check this theory out...
Here's our test file:
https://living.video/direct5c.php?item=hphfawsley

**Comment by Soviut** [comment]
I too have noticed the poster re-displaying while seeking and even paused. The solution I'll be using for now is to just set a style to hide it once the `vjs-has-started` class is applied to the container.

**Comment by gkatsev** [comment]
Fixed by #3337.
